apps/vendors/views.py

```
@@ -0,0 +1,47 @@
+
+    def get_queryset(self):
+        user = self.request.user
+        queryset = Vendor.objects.select_related('user')
+        if user.is_staff or user.is_superuser:
+            return queryset
+        if hasattr(user, 'vendor_profile'):
+            return queryset.filter(pk=user.vendor_profile_id)
+        return queryset.filter(is_active=True, is_verified=True)
+
+    @action(detail=True, methods=['post'])
+    def verify(self, request, pk=None):
```

Ruff (0.14.0)

Unused method argument: `pk`

(ARG002)